mithril-client/src/services/snapshot.rs

```diff
@@ -472,40 +531,65 @@ mod tests {
     async fn test_download_snapshot_ok() {
         let test_path = std::env::temp_dir().join("test_download_snapshot_ok");
         let _ = std::fs::remove_dir_all(&test_path);
-        let mut http_client = MockAggregatorHTTPClient::new();
-        http_client.expect_probe().returning(|_| Ok(()));
-        http_client
-            .expect_download()
-            .returning(move |_, _, _| Ok(()))
-            .times(1);
-        http_client.expect_get_content().returning(|_| {
-            let mut message = CertificateMessage::dummy();
-            message.signed_message = message.protocol_message.compute_hash();
-            let message = serde_json::to_string(&message).unwrap();
 
-            Ok(message)
-        });
+        let (http_client, certificate_verifier, digester) =
+            get_mocks_for_snapshot_service_configured_to_make_download_succeed();
+
         let mut builder = get_dep_builder(Arc::new(http_client));
-        let mut certificate_verifier = MockCertificateVerifierImpl::new();
-        certificate_verifier
-            .expect_verify_certificate_chain()
-            .returning(|_, _, _| Ok(()))
-            .times(1);
         builder.certificate_verifier = Some(Arc::new(certificate_verifier));
-        builder.immutable_digester = Some(Arc::new(DumbImmutableDigester::new(
-            "snapshot-digest-123",
-            true,
-        )));
-        let snapshot = FromSnapshotMessageAdapter::adapt(get_snapshot_message());
+        builder.immutable_digester = Some(Arc::new(digester));
         let snapshot_service = builder.get_snapshot_service().await.unwrap();
 
+        let snapshot = FromSnapshotMessageAdapter::adapt(get_snapshot_message());
+        build_dummy_snapshot(
+            "digest-10.tar.gz",
+            "1234567890".repeat(124).as_str(),
+            &test_path,
+        );
+
         let (_, verifier) = setup_genesis();
         let genesis_verification_key = verifier.to_verification_key();
+
+        let filepath = snapshot_service
+            .download(
+                &snapshot,
+                &test_path,
+                &key_encode_hex(genesis_verification_key).unwrap(),
+                ProgressDrawTarget::hidden(),
+            )
+            .await
+            .expect("Snapshot download should succeed.");
+        assert!(
+            filepath.is_dir(),
+            "Unpacked location must be in a directory."
+        );
+        assert_eq!(Some(OsStr::new("db")), filepath.file_name());
+    }
+
+    #[tokio::test]
+    async fn test_download_snapshot_ok_add_clean_file_allowing_node_bootstrap_speedup() {
```

Comment:
Do we have a test checking _not adding_ the 'clean' file still works? I was under the impression the could would always try to create it.

Reply:
I would be nice bust costly , to do that cleanly I need to extract the code that insert the `clean` file in a separated service to mock it. For such a small part I don't think that's worth the increased complexity.